### PR TITLE
PWN-6858 - Off-ramp. New sell amount widget

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6858

## Description of Work

small step to other fiat support in app, i hope you don't mind

### CODE REVIEW

File: core/src/main/java/org/p2p/core/model/CurrencyMode.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.p2p.core.utils.Constants
+import org.p2p.core.utils.Constants.FIAT_FRACTION_LENGTH
+import org.p2p.core.token.Token as UserToken
+
+sealed interface CurrencyMode {
+    data class Token constructor(val symbol: String, val fractionLength: Int) : CurrencyMode {
+        constructor(token: UserToken) : this(token.tokenSymbol, token.decimals)
+    }
+
+    sealed class Fiat(val fiatSymbol: String, val fiatAbbreviation: String) : CurrencyMode {
```
Comment (gslevinkov) [inline core/src/main/java/org/p2p/core/model/CurrencyMode.kt:12]: or maybe we need to create another model class like `Token.Active` with info and use it here, whatcha think?

File: core/src/main/java/org/p2p/core/utils/AmountExtensions.kt [hunk 2]
```diff
@@ -66,7 +65,7 @@ fun BigDecimal.toUsd(token: Token): BigDecimal? =
     token.rate?.let { this.multiply(it).scaleShort() }
 
 // case: 1000.023000 -> 1 000.02
-fun BigDecimal.formatUsd(): String = formatWithDecimals(USD_DECIMALS)
+fun BigDecimal.formatFiat(): String = formatWithDecimals(FIAT_FRACTION_LENGTH)
```
Comment (gslevinkov) [inline core/src/main/java/org/p2p/core/utils/AmountExtensions.kt:68]: same 2

File: ui-kit/src/main/res/layout/widget_sell_input_amount.xml [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        android:ellipsize="middle"
+        android:gravity="start"
+        android:textAppearance="@style/UiKit.TextAppearance.Regular.Label1"
+        android:textColor="@color/text_mountain"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:text="1 SOL ≈ 12.85 USD" />
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (NikolaiEmelianov) [inline ui-kit/src/main/res/layout/widget_sell_input_amount.xml:27]: why nested ConstraintLayout is needed? use barrier or guideline

Comment (gslevinkov) [reply ui-kit/src/main/res/layout/widget_sell_input_amount.xml:27]: to group white block in one container, barrier or guideline won't help

File: core/src/main/java/org/p2p/core/model/CurrencyMode.kt (hunk 1 above)
Comment (1sakov) [reply core/src/main/java/org/p2p/core/model/CurrencyMode.kt:12]: this lgtm

**Review by 1sakov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**